pkg/tms/tms.go

```diff
@@ -10,14 +10,44 @@ import (
 	"net/http"
 	"net/url"
 	"os"
+	"sort"
 	"strings"
 
+	"github.com/SAP/jenkins-library/pkg/command"
 	piperHttp "github.com/SAP/jenkins-library/pkg/http"
 	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/ghodss/yaml"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	DEFAULT_TR_DESCRIPTION = "Created by Piper"
+)
+
+type TmsUtils interface {
+	command.ExecRunner
+
+	FileExists(filename string) (bool, error)
+	FileRead(path string) ([]byte, error)
+
+	// Add more methods here, or embed additional interfaces, or remove/replace as required.
+	// The tmsUploadUtils interface should be descriptive of your runtime dependencies,
+	// i.e. include everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type uaa struct {
+	Url          string `json:"url"`
+	ClientId     string `json:"clientid"`
+	ClientSecret string `json:"clientsecret"`
+}
+
+type serviceKey struct {
+	Uaa uaa    `json:"uaa"`
+	Uri string `json:"uri"`
+}
+
```

Comment:
Could the `xsuaa` package used for `serviceKey` and `token` related stuff?

Reply:
These structures are specific to each application's service broker. xsuaa.AuthToken can be reused